doc/release-notes.md

```diff
@@ -85,6 +85,7 @@ Wallet
 ------
 
 - The wallet now by default uses bech32 addresses when using RPC, and creates native segwit change outputs.
+- The way that output trust was computed has been fixed in #16766, which impacts confirmed/unconfirmed balance status and coin selection.
```

Comment:
nit: maybe precise how the trust computation has been changed "to evaluate trustiness of outpoint spent by parent and so recursively"

Reply:
I'm not sure what level of detail is best for the release note; will leave up to maintainer discretion. Since it can be nuanced, if someone is relying on this it's best that they read the PR.